firmware/application/src/app_main.c

```diff
@@ -152,12 +161,41 @@ static void timer_button_event_handle(void *arg) {
             if (settings_get_button_press_config('b') != SettingsButtonDisable) {
                 NRF_LOG_INFO("BUTTON_LEFT"); // Button B?
                 m_is_b_btn_press = true;
+                m_last_btn_press = app_timer_cnt_get();
             }
         }
         if (pin == BUTTON_2) {
             if (settings_get_button_press_config('a') != SettingsButtonDisable) {
                 NRF_LOG_INFO("BUTTON_RIGHT"); // Button A?
                 m_is_a_btn_press = true;
+                m_last_btn_press = app_timer_cnt_get();
+            }
+        }
+    }
+
+    if (nrf_gpio_pin_read(pin) == 0) {
+        uint32_t now = app_timer_cnt_get();
+        uint32_t ticks = app_timer_cnt_diff_compute(now, m_last_btn_press);
+
+        uint32_t time = ticks * ((APP_TIMER_CONFIG_RTC_FREQUENCY + 1 ) * 1000 ) / APP_TIMER_CLOCK_FREQ;
+
+        bool is_long_press = time > 1000;
```

Comment:
sorry one more comment :)
I was about to suggest to create a macro as this can be useful elsewhere, but there is one for the inverse so you could replace by
```suggestion
        bool is_long_press = time > APP_TIMER_TICKS(1000);
```